plugins/dell-k2/fu-dell-k2-rtshub-firmware.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include "fu-dell-k2-common.h"
+#include "fu-dell-k2-rtshub-firmware.h"
+
+#define DOCK_RTSHUB_GEN2_VERSION_OFFSET 0x7F52
+#define DOCK_RTSHUB_GEN1_VERSION_OFFSET 0x7FA6
+#define DOCK_RTSHUB_GEN1_VID_OFFSET	0x7FA8
+#define DOCK_RTSHUB_GEN1_PID_OFFSET	0x7FAA
+
+struct _FuDellK2RtshubFirmware {
+	FuFirmwareClass parent_instance;
+	guint16 pid;
+};
+
+G_DEFINE_TYPE(FuDellK2RtshubFirmware, fu_dell_k2_rtshub_firmware, FU_TYPE_FIRMWARE)
+
+static gchar *
+fu_dell_k2_rtshub_firmware_convert_version(FuFirmware *firmware, guint64 version_raw)
+{
+	return fu_version_from_uint32_hex(version_raw, fu_firmware_get_version_format(firmware));
+}
+
+static void
+fu_dell_k2_rtshub_firmware_export(FuFirmware *firmware,
+				  FuFirmwareExportFlags flags,
+				  XbBuilderNode *bn)
+{
+	FuDellK2RtshubFirmware *self = FU_DELL_K2_RTSHUB_FIRMWARE(firmware);
+	fu_xmlb_builder_insert_kx(bn, "product_id", self->pid);
+}
+
+static gboolean
+fu_dell_k2_rtshub_firmware_set_offset(GInputStream *stream,
+				      guint16 *version_offset,
+				      guint16 *pid_offset,
+				      GError **error)
+{
+	guint16 vid_raw = 0;
+
+	if (!fu_input_stream_read_u16(stream,
+				      DOCK_RTSHUB_GEN1_VID_OFFSET,
+				      &vid_raw,
+				      G_BIG_ENDIAN,
+				      error))
+		return FALSE;
+
+	if (vid_raw == DELL_VID) {
+		*version_offset = (guint16)DOCK_RTSHUB_GEN1_VERSION_OFFSET;
+		*pid_offset = (guint16)DOCK_RTSHUB_GEN1_PID_OFFSET;
+		return TRUE;
+	}
+
+	*version_offset = (guint16)DOCK_RTSHUB_GEN2_VERSION_OFFSET;
+	return TRUE;
+}
+static gboolean
+fu_dell_k2_rtshub_firmware_parse(FuFirmware *firmware,
+				 GInputStream *stream,
+				 gsize offset,
+				 FwupdInstallFlags flags,
+				 GError **error)
+{
+	FuDellK2RtshubFirmware *self = FU_DELL_K2_RTSHUB_FIRMWARE(firmware);
+	guint16 version_raw = 0;
+	guint16 version_offset = 0;
+	guint16 pid_raw = 0;
+	guint16 pid_offset = 0;
+	g_autofree gchar *version_str = NULL;
```

Comment:
variable is unused, can be dropped

Reply:
Fixed, thanks.  Weirdly CI x86_64 build is still failing.